### PR TITLE
Clarify comment on parsing new lines and factor out function

### PR DESCRIPTION
**Pull request by luhring**
- Clarifies comment on why an empty line in a `METADATA` or `PKG_INFO` file means that we cease parsing, with a reference to the spec
- Factors out field continuation logic into new function (I wasn't originally intending to do this, but the above comment clarification pushed the function's length past the linter's threshold, and it seemed like a good time to do some small cleanup.)

(This was investigated originally due to https://github.com/anchore/syft/issues/280.)

Signed-off-by: Dan Luhring <dan.luhring@anchore.com>

### CODE REVIEW

**Review by alfredodeza — APPROVED** [review]

File: syft/cataloger/python/parse_wheel_egg_metadata.go [hunk 1]
```diff
@@ -71,8 +70,23 @@ func parseWheelOrEggMetadata(path string, reader io.Reader) (pkg.PythonPackageMe
...
-	sitePackagesRoot := filepath.Clean(filepath.Join(filepath.Dir(path), ".."))
-	metadata.SitePackagesRootPath = sitePackagesRoot
+	metadata.SitePackagesRootPath = filepath.Clean(filepath.Join(filepath.Dir(path), ".."))
 
 	return metadata, nil
 }
+
+// handleFieldBodyContinuation returns the updated value for the specified field after processing the specified line.
+// If the continuation cannot be processed, it returns an error.
+func handleFieldBodyContinuation(key, line string, fields map[string]string) (string, error) {
```
Comment (wagoodman) [inline syft/cataloger/python/parse_wheel_egg_metadata.go:80]: I take it the linter rules nudged this above the function length due to added comments?...

**Review by wagoodman — APPROVED** [review]

File: syft/cataloger/python/parse_wheel_egg_metadata.go (hunk 1 above)
Comment (luhring) [reply syft/cataloger/python/parse_wheel_egg_metadata.go:80]: @wagoodman Yep, exactly — see the PR description

Comment (wagoodman) [reply syft/cataloger/python/parse_wheel_egg_metadata.go:80]: if only there was a spot in a PR where I could read about context... 🤔 ...GREAT SCOTT!! When did they add PR descriptions??!! (...my bad)

Comment (luhring) [reply syft/cataloger/python/parse_wheel_egg_metadata.go:80]: great scott

**Pull request merged**